src/request/components/request-detail-panel-request.tsx

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+/*tslint:disable:no-var-requires */
+const messageProcessor = require('../util/request-message-processor');
+/*tslint:enable:no-var-requires */
+
+import _ = require('lodash');
+import React = require('react');
+
+const getPayloads = (function() {
+    const getItem = messageProcessor.getTypePayloadItem;
+
+    const options = {
+        'web-response': getItem,
+        'web-request': getItem
+    };
+
+    return function(request) {
+        return messageProcessor.getTypeStucture(request, options);
+    };
+})();
+
+class RequestUrl extends React.Component<{url: string}, {}> {
```

Comment:
Yes, once you declare the types for the properties passed to the component, you then get IntelliSense both when using those properties within the component as well as when setting the attributes on the component (from another component).  It works pretty slick.